docs/guides/bearer_token/bearer_token_guide.md

```diff
@@ -0,0 +1,68 @@
+---
+uid: Guides.BearerToken
+title: Working with Bearer token
+---
+
+# Working with Bearer token
+
+Some endpoints in Discord API require a Bearer token, which can be obtained through [OAuth2 flow](https://discord.com/developers/docs/topics/oauth2). Discord.Net allows you to interact with these endopoints using [DiscordRestClient].
+
+## Initializing a new instance of the client
+[!code-csharp[Initialize DiscordRestClient](samples/rest_client_init.cs)]
+
+## Getting current user
+
+[DiscordRestClient] gets the current user when `LoginAsync()` is called. The user object can be found in `CurrentUser` property.
+
+If you need to fetch the user again `GetGetCurrentUserAsync()` method can be used.
+
+[!code-csharp[Get current user](samples/current_user.cs)]
+
+> [!NOTE]
+> Some properies might be `null` depending on which scopes user authed your app with.
+> For example: `email` scope is required to fetch current user's email address.
+
+## Fetching current user's guilds
+
+`GetGuildSummariesAsync()` method is used to fetch current user's guilds. Since it returns an `IAsyncEnumerable` you need to call `FlattenAsync()` to get a plain `IEnumerable` containing [RestUserGuild] objects.
```

Comment:
```suggestion
The `GetGuildSummariesAsync()` method is used to fetch current user's guilds. Since it returns an `IAsyncEnumerable` you need to call `FlattenAsync()` to get a plain `IEnumerable` containing [RestUserGuild] objects.
```